service/src/main/java/com/codeforcommunity/processor/TeamsProcessorImpl.java

```diff
@@ -280,4 +295,47 @@ public GetUserTeamsResponse getUserTeams(JWTData userdata) {
     }
     return teamMembers;
   }
+
+  @Override
+  public void transferOwnership(JWTData userData, TransferOwnershipRequest request) {
+    int ownerId = userData.getUserId();
+
+    UserTeamRecord userTeamRecordOwner =
+        db.selectFrom(USER_TEAM)
+            .where(USER_TEAM.USER_ID.eq(ownerId))
+            .and(USER_TEAM.TEAM_ID.eq(request.getTeamId()))
+            .fetchOneInto(UserTeamRecord.class);
+
+    if (userTeamRecordOwner == null || userTeamRecordOwner.getTeamRole() != TeamRole.LEADER) {
+      throw new TeamLeaderOnlyRouteException(request.getTeamId());
+    }
+
+    Optional<UsersRecord> newOwnerOptional =
+        db.selectFrom(USERS).where(USERS.EMAIL.eq(request.getNewOwnerEmail())).fetchOptional();
+
+    UsersRecord newOwner =
+        newOwnerOptional.orElseThrow(
+            () -> new UserDoesNotExistException(request.getNewOwnerEmail()));
+
+    boolean userOnTeam =
+        db.fetchExists(
```

Comment:
Let's get the record here like we are for the leader and then just update that to keep things consistent

Reply:
I had that originally, but I had to remove it since it's not using `UpdatableRecord`. I'll change this if you think adding the PK would be good.